grakn-graql/src/main/java/ai/grakn/graql/internal/printer/HALPrinter.java

```diff
@@ -36,9 +39,39 @@
     }
 
     @Override
-    public Json graqlString(boolean inner, Concept concept) {
-        String json = HALBuilder.renderHALConceptData(concept, 0, keyspace,0, limitEmbedded);
+    public Json graqlString(boolean inner, Answer answer) {
+
+        Json json = Json.object();
+
+        answer.map().forEach((Var key, Concept value) -> {
+            String keyString = key.getValue();
+            json.set(keyString, halString(value, isInferred(key, value, answer)));
+        });
+
+        return json;
+    }
+
+    public Json halString(Concept concept, boolean inferred) {
+        String json = HALBuilder.renderHALConceptData(concept, inferred, 0, keyspace, 0, limitEmbedded);
         return Json.read(json);
     }
 
+    // TODO: Add 'inferred' flag to concept
+    private boolean isInferred(Var key, Concept concept, Answer answer) {
+        if (key == null || answer.getExplanation().isEmpty()) return false;
+
+        //TO-DO add support for attributes
```

Comment:
🐘

Reply:
Also we might not need this one, not sure, I will discuss it with Kasper. But definitely the future